### PR TITLE
[JIT] ARM64 - Combine 'neg' and 'cmp' to 'cmn'

### PR DESCRIPTION
**Pull request by TIHan**
## Description

Part of the work for https://github.com/dotnet/runtime/issues/68028

Handles the following ARM64 multi-op optimizations, with examples:

1. `neg`, `cmp` => `cmn`
```asm
neg x1, x1
cmp x1, x0
```
to
```asm
cmn x0, x1
```
---
2. `lsr/asr/lsl`, `neg` => `neg` (shifted)
```asm
lsr/asr/lsl w0, w0, #1
neg         w0, w0
```
to
```asm
neg w0, w0, LSR/ASR/LSL #1
```
---
3. `lsr/asr/lsl`, `neg`, `cmp` => `cmn` (shifted)
```asm
lsr/asr/lsl w1, w1, #2
neg         w1, w1
cmp         w1, w0
```
to
```asm
cmn w0, w1, LSR/ASR/LSL #2
```

### CODE REVIEW

**Comment by TIHan** [comment]
@dotnet/jit-contrib This is ready. @BruceForstall // cc - pending CI

File: src/coreclr/jit/lower.cpp [hunk 1]
```diff
@@ -3916,6 +3916,17 @@ bool Lowering::TryLowerConditionToFlagsNode(GenTree* parent, GenTree* condition,
...
+        else if (optimizing && relop->OperIs(GT_EQ, GT_NE, GT_LT, GT_LE, GT_GE, GT_GT) &&
+                 (IsContainableUnaryOrBinaryOp(relop, relop->gtGetOp1()) ||
+                  IsContainableUnaryOrBinaryOp(relop, relop->gtGetOp2())))
+        {
+            ContainCheckCompare(relop);
+            relop->SetOper(GT_CMP);
+            relop->gtType = TYP_VOID;
+            relop->gtFlags |= GTF_SET_FLAGS;
+        }
+#endif // TARGET_ARM64
```
Comment (BruceForstall) [inline src/coreclr/jit/lower.cpp:3929]: This seems a bit odd. It does exactly what would happen in the `else` case below, unconditionally, except it also does the `ContainCheckCompare`, and also, it doesn't move the `relop` down, as specified in the function contract.

Don't we call ContainCheckCompare in LowerCompare() before we get here?

**Review by BruceForstall — COMMENTED** [review]
One question

File: src/coreclr/jit/lower.cpp (hunk 1 above)
Comment (TIHan) [reply src/coreclr/jit/lower.cpp:3929]: `LowerCompare` does call `ContainCheckCompare` and `LowerCompare` should have already been called by now. I will check this.

Comment (TIHan) [reply src/coreclr/jit/lower.cpp]: Ok, it turns out I didn't even need to have that code there. 😒

**Comment by BruceForstall** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=246470&view=ms.vss-build-web.run-extensions-tab)

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**